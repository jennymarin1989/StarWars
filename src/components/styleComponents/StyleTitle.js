import styled from 'styled-components';

const StyleTitle = styled.text`
    font-size: 0.8em;
    color: red;
    font-family: verdana;
    text-align: center;
`;

export default StyleTitle;