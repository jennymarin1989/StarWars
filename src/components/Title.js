import React from 'react';
import StyleTitle from './styleComponents/StyleTitle';

const Title = (props) => {

  return (

    <div>
      <StyleTitle>
         <h1 style={{ backgroundColor: 'red', display: 'inline' }}> { props.header } </h1>
      </StyleTitle>
    </div>  

  )

}

export default Title;