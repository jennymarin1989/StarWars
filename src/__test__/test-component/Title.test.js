import React from 'react';
import { shallow, mount } from 'enzyme';
import Title  from '../../components/Title';
import StyleTitle from '../../components/styleComponents/StyleTitle';

describe ('Title', () => {

  let title;

  beforeEach(()=>{
    title = shallow(<Title />);
  })

  describe( 'renders components correctly', () => {
    it('renders Title component correctly', () =>{
     expect(title).toMatchSnapshot(); 
    });

    it ('passes the props from parent to child component', () => {
      let title = mount(<Title header={"LEGO"}/>)
      expect(title.props().header).toBe("LEGO")
    });

    it('renders style component correctly', () =>{
      expect(title.find(StyleTitle)).toHaveLength(1);
    })
  })
})