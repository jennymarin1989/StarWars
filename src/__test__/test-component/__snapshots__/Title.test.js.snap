// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders components correctly renders Title component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <styled.text>
        <h1
                style={
                        Object {
                                "backgroundColor": "red",
                                "display": "inline",
                              }
                }
        >
                 
                 
        </h1>
</styled.text>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h1
          style={
                    Object {
                              "backgroundColor": "red",
                              "display": "inline",
                            }
          }
>
           
           
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            undefined,
            " ",
          ],
          "style": Object {
            "backgroundColor": "red",
            "display": "inline",
          },
        },
        "ref": null,
        "rendered": Array [
          " ",
          undefined,
          " ",
        ],
        "type": "h1",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <styled.text>
          <h1
                    style={
                              Object {
                                        "backgroundColor": "red",
                                        "display": "inline",
                                      }
                    }
          >
                     
                     
          </h1>
</styled.text>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            style={
                        Object {
                                    "backgroundColor": "red",
                                    "display": "inline",
                                  }
            }
>
             
             
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              undefined,
              " ",
            ],
            "style": Object {
              "backgroundColor": "red",
              "display": "inline",
            },
          },
          "ref": null,
          "rendered": Array [
            " ",
            undefined,
            " ",
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
